crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -953,31 +953,11 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
                     self.markers.as_ref(),
                     self.requires_python.as_ref(),
                     markers,
-                );
-
-                let dependencies = match dependencies {
-                    Ok(dependencies) => dependencies,
-                    Err(err) => {
-                        return Ok(Dependencies::Unavailable(
-                            UnavailableVersion::ResolverError(uncapitalize(err.to_string())),
-                        ));
-                    }
-                };
```

Comment:
@charliermarsh Do you know what this was about? It's only in this branch (in the non-root package branch we short-circuit on error) and the errors we're handling here look like they should be raised.

Reply:
Why should they be raised? You don't think we should backtrack in these cases?

Reply:
As far as i can tell, the current implementation means that if there is a direct dependency (a root package dependency) where `PubGrubSpecifier::try_from` fails or which uses a banned URL, we mark those dependencies as unavailable, so the root package becomes unavailable(?). For regular dependencies, we already raise those errors directly using `from_requirements(...)?`.

This code looks like an artifact to me, but going through the git history unfortunately didn't clear this up either.

Reply:
It looks like there are _some_ errors that we'd want to recover from though, like `InvalidVersion` or perhaps `PubGrubSpecifierError`?

Reply:
I don't think they are recoverable, as far as i can tell they lead to the root package being rejected in the next step, so you get the same error except it's wrapped in a message about the root package being unavailable.

Reply:
We also do this only for the root package, which looks wrong.

Reply:
Where are they rejected? Note that this does not return an error; it returns `Ok(Dependencies::Unavailable)`.

Reply:
This code is unreachable (both error cases are raised earlier), so i'm removing it